### PR TITLE
TapToEdit - add useEffect for initialValue handling

### PR DESCRIPTION
**Pull request by hchamorro**
### **User description**
Issue:
https://github.com/orgs/FlourishHealth/projects/1/views/17?pane=issue&itemId=45864486

add useEffect to handle initial value and update initial value on save

___

### **PR Type**
Enhancement

___

### **Description**
- Added `useEffect` to set the initial value after the initial render in `TapToEdit`.
- Updated `initialValue` state when the `onSave` function is called.
- Imported `useEffect` from React to manage side effects.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>TapToEdit.tsx</strong><dd><code>Add `useEffect` for initial value handling in TapToEdit</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/TapToEdit.tsx

<li>Added <code>useEffect</code> to handle initial value setting.<br> <li> Updated <code>initialValue</code> state on save.<br> <li> Imported <code>useEffect</code> from React.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/576/files#diff-5c106f38db848acb6a5d382743af81c919163754380b895781575765bc292ecd">+9/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/ui/src/TapToEdit.tsx [hunk 1]
```diff
@@ -97,12 +97,21 @@ export const TapToEdit = ({
   description: propsDescription,
   showDescriptionAsTooltip = false,
   onlyShowDescriptionWhileEditing = true,
+  instanceId,
```
Comment (swanka) [inline packages/ui/src/TapToEdit.tsx:100]: What are we planning to use as the instanceid from the parent component? Just the document id?

Comment (hchamorro) [reply packages/ui/src/TapToEdit.tsx:100]: instanceId is like react key. It can be any unique identifier. Will update comment in common

Comment (hchamorro) [reply packages/ui/src/TapToEdit.tsx]: alright sorry about the confusion this is all set to go now

**Review by swanka — APPROVED** [review]
Nice!

**Pull request merged**